File: factory/src/main/java/com/zuzex/factory/service/OrderService.java
package com.zuzex.factory.service;

import com.zuzex.factory.model.Order;

import java.util.List;

public interface OrderService {

    List<Order> findAllOrders();

    Order findOrderById(Long id);

    Order createNewOrderByCarIdAndDescription(Long carId, String description);

    Order assembleOrderById(Long id);

    Order deliverOrderById(Long id);

    Order revertOrderStatusById(Long id);
}

File: car-factory/src/main/kotlin/com/zuzex/carfactory/service/OrderService.kt
package com.zuzex.carfactory.service

import com.zuzex.carfactory.model.Order

interface OrderService {

    fun findAllOrders(): List<Order>

    fun findOrderById(id: Long): Order

    fun createNewOrderByCarIdAndDescription(carId: Long, description: String?): Order

    fun changeOrderStatusById(id: Long, action: String): Order

    fun revertOrderStatusById(id: Long): Order
}
